### DISCUSSION TITLE
wallet fees problem

### DISCUSSION THREAD

**Discussion by dxteam2024** [dogecoin/dogecoin#2848] [Help needed]
Why in Testnet the fees is not stable, sometimes stay high sometimes stay low. What do I miss? Just look at the records. 

https://chain.so/tx/DOGETEST/fe84c2df64b57546ee54b34bd59ec1ca3046f08e206ff6f30de8dc7ea9e93b9c   high fees
https://chain.so/tx/DOGETEST/a1bdfbf875df1ca6c387eef19a298f0f6258eba37c3a2ba00a44c28a2e3e69e2   high fees
https://chain.so/tx/DOGETEST/a4329c27ac7adb11956d21838802a20eca0cd3f47946a9692136b8d6341f2a8a low fees 

 I set the 0.01 doge /kb

**Comment by didlie** [comment]
this is a setting that can be adjusted in your wallet. You can offer whatever fee you want for your transactions.

**Comment by patricklodder** [comment]
@aliray which wallet are you using?

**Reply by dxteam2024** [reply]
Thanks for the answer, I try to dev a wallet & I fixed the problem, It's 1 Doge dust limit, the tools I used ignore the change output.

**Reply by dxteam2024** [reply]
I read the fees doc, the wallet dust limit  1doge now is suitable? Can set a lower dust limit for my wallet?

**Reply by patricklodder** [reply]
You can set it lower but not every mining pool supports it yet, so you may see some delays in your transaction getting confirmed. 

You can lower it - possibly until 0.01 DOGE - with `-discardthreshold=0.xx`. We currently don't have nodes exchange their dust thresholds, but looking at feefilters and versions of 2300 peers my nodes connect to, it's unlikely that anything under 0.01 will propagate far enough to get mined, but about half the network and some miners allow 0.01 DOGE outputs.

**Comment by dxteam2024** [comment]
<html>
<body>


inputs | outputs | kb | fee | fee per kb | % of tx
-- | -- | -- | -- | -- | --
1 | 1 | 192 | 1 DOGE | 5.21 DOGE | 4%
1 | 2 | 226 | 1 DOGE | 4.42 DOGE | 36%
2 | 1 | 340 | 1 DOGE | 2.94 DOGE | 11%
2 | 2 | 374 | 1 DOGE | 2.67 DOGE | 5%
1 | 24 | 974 | 1 DOGE | 1.03 DOGE | -


</body>
</html>

2 inputs 1 output should not be 350 why 340? base the table listed, 1 input = 158  , 1 output = 34,  Am I wrong?

**Reply by dxteam2024** [reply]
table from this https://github.com/dogecoin/dogecoin/discussions/2347#discussion-3433486

**Reply by patricklodder** [reply]
Input size is 148 bytes, you're adding per-transaction overhead to the inputs.

- 10 bytes overhead (version (4), number of inputs (1), number of outputs (1), lock time (4))
- 2 inputs of 148 bytes
- 1 outputs of 34 bytes

10 + 148 + 148 + 34 = 340

**Reply by dxteam2024** [reply]
What about relay fees, Should i add the relay fees to the network fees? ep: send 1 doge to somebody, use 1 input + 1 output, so the size was 192 and fees  = (192 /1000)*0.01  + (192 /1000) * 0.001（fee for relay）. Will it affect the speed of confirmation?

**Reply by patricklodder** [reply]
no. Relay fee is the minimum fee rate that the network enforces, it's the same calculation, only enforced at a lower rate because it allows the network to collectively decide on fees without needing developer interference.

So as long as you keep your output amounts higher than the soft dust limit, the calculation is rather straightforward:

`fee = bytes * feerate / 1000`

so in case of a 192 tx and a 0.01 DOGE/kb fee rate, you would get: `192 * 0.01 / 1000 = 0.00192`

The difference between the original fee proposal and the [implemented defaults](https://github.com/dogecoin/dogecoin/blob/master/doc/fee-recommendation.md) is that the reduction from the main proposal is implemented while dropping the rounding from the alternative. So it's cheaper than planned now, but also simpler.

**Reply by dxteam2024** [reply]
Thanks for the patients and answer！

**Comment by emperor999-afk** [comment]
It's Been fixed
